src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -297,10 +301,96 @@ private ImmutableArray<ISourceFile> CloseCompilationInternal(DocumentUri documen
             }
         }
 
-        private void SendLinterStateTelemetryOnBicepFileOpen(RootConfiguration configuration)
+        private void SendTelemetryOnBicepFileOpen(DocumentUri documentUri, RootConfiguration configuration, ImmutableHashSet<ISourceFile> sourceFiles, IEnumerable<Diagnostic> diagnostics)
         {
+            // Telemetry on linter state on bicep file open
             var telemetryEvent = GetLinterStateTelemetryOnBicepFileOpen(configuration);
             TelemetryProvider.PostEvent(telemetryEvent);
+
+            // Telemetry on open bicep file and the referenced modules
+            telemetryEvent = GetTelemetryAboutSourceFiles(documentUri.ToUri(), sourceFiles, diagnostics);
+
+            if (telemetryEvent is not null)
+            {
+                TelemetryProvider.PostEvent(telemetryEvent);
+            }
+        }
+
+        public BicepTelemetryEvent? GetTelemetryAboutSourceFiles(Uri uri, ImmutableHashSet<ISourceFile> sourceFiles, IEnumerable<Diagnostic> diagnostics)
+        {
+            var mainFile = sourceFiles.First(x => x.FileUri == uri) as BicepFile;
+
+            if (mainFile is null)
+            {
+                return null;
+            }
+
+            Dictionary<string, string> properties = GetTelemetryPropertiesForMainFile(mainFile, diagnostics);
+
+            var referencedFiles = sourceFiles.Where(x => x.FileUri != uri);
+            var propertiesFromReferencedFiles = GetTelemetryPropertiesForReferencedFiles(referencedFiles);
+
+            properties = properties.Concat(propertiesFromReferencedFiles).ToDictionary(s => s.Key, s => s.Value);
+
+            return BicepTelemetryEvent.CreateBicepFileOpen(properties);
+        }
+
+        private Dictionary<string, string> GetTelemetryPropertiesForMainFile(BicepFile bicepFile, IEnumerable<Diagnostic> diagnostics)
+        {
+            Dictionary<string, string> properties = new();
+
+            var declarationsInMainFile = bicepFile.ProgramSyntax.Declarations;
+            properties.Add("Modules", declarationsInMainFile.Count(x => x is ModuleDeclarationSyntax).ToString());
+            properties.Add("Parameters", declarationsInMainFile.Count(x => x is ParameterDeclarationSyntax).ToString());
+            properties.Add("Resources", declarationsInMainFile.Count(x => x is ResourceDeclarationSyntax).ToString());
+            properties.Add("Variables", declarationsInMainFile.Count(x => x is VariableDeclarationSyntax).ToString());
+
+            var localPath = bicepFile.FileUri.LocalPath;
+
+            if (File.Exists(localPath))
```

Comment:
I believe File.Exists can throw (e.g. bad path, or user doesn't have permissions to the path).